---
title: "*Clostridium difficile* differentially alters the structure and metabolism of distinct cecal microbiomes to promote sustained colonization during infection"
csl: elife.csl
output:
 word_document:
  keep_md: true
  reference_docx: manuscript_format.docx
bibliography: references.bib
---

**Authors:** Matthew L. Jenior, Jhansi L. Leslie, Vincent B. Young, and Patrick D. Schloss^\*^


###Abstract

*Clostridium difficile* has become the most common single cause of hospital-acquired infection over the last decade in the United States. Susceptibility is primarily associated with previous exposure to antibiotics, which compromise the structure and function of the gut bacterial community. Furthermore, specific antibiotic classes correlate more strongly with recurrent or persistent *C. difficile* infection. We used a murine model of infection to explore the effect of distinct antibiotic classes on sustained *C. difficile* colonization, as well as the impact of infection on community-level gene expression and metabolism 18 hours post-infection. Utilizing untargeted metabolomic analysis, we found that *C. difficile* infection has larger impacts on the metabolic activity of the microbiota across cefoperazone and streptomycin-pretreated mice, which become persistently colonized compared to clindamycin-pretreated mice that cleared the infection within 8 days. Through metagenome-enabled metatranscriptomics, we observed that the infected microbial communities were enriched in pathways associated with amino acid metabolism and particularly in non-dominant species relative to mock-infected controls. Conversely, in mice pretreated with clindamycin, the effect of infection on the microbiota was only detectable in changes to the community structure but not in metabolic activity or gene expression. Our results suggest that *C. difficile* is able to restructure the nutrient-niche landscape in certain gut environments in order to promote persistent infection.


###Introduction

One of the many beneficial functions provided by the indigenous gut bacterial community is its ability to protect the host from infection by pathogens [@Vollaard1994]. This attribute, termed colonization resistance, is one of the main mechanisms that protect healthy individuals from the gastrointestinal pathogen *Clostridium difficile* [@Freter1955;@Fekety1979;@Britton2012]. *C. difficile* infection (CDI) is responsible for most cases of antibiotic-associated colitis, a toxin-mediated diarrheal disease that has dramatically increased in prevalence over the last 10 years. There are an estimated 453,000 cases of CDI resulting in 29,000 deaths in the US annually [@Lessa2015]. Antibiotics are a major risk factor for CDI and are thought to increase susceptibility to CDI as they disrupt the gut bacterial community structure; however, it is still unclear what specific changes to the microbiota contribute to this susceptibility [@Antonopoulos2009;@Buffie2012]. While most classes of antibiotics have been associated with initial susceptibility to CDI, fluoroquinolones, clindamycin, and cephalosporins are linked to increased risk of recurrent or persistent infection [@Thomas2003;@Brown2013;@Bignardi1998]. This raises questions about the groups of bacteria that are differentially impacted by certain therapies and what that ultimately translates to in duration or severity of the infection.

Associations between the membership and functional capacity of the microbiota as measured by the metabolic output suggest that antibiotics increase susceptibility by altering the nutrient milieu in the gut to one that favors *C. difficile* metabolism [@Antunes2011;@Jump2014;@Theriot2014]. One hypothesis is that *C. difficile* colonization resistance is driven by competition for growth substrates by an intact community of metabolic specialists. This has been supported by animal model experiments over the past several decades [@Wilson1988;@Sambol2002;@Perez-Cobas2014]. This line of reasoning has been carried through to the downstream restoration of colonization resistance with the application of fecal microbiota transplant (FMT). Although a person's microbiota may not return to its precise original state following FMT, it is hypothesized that the functional capacity of the new microbiota is able to outcompete *C. difficile* for resources and clear the infection [@Zaura2015;@Theriot2014]. 

Leveraging distinct antibiotic treatment regimens in a murine model of CDI [@Schubert2015], we and others have shown that *C. difficile* adapts its physiology to the distinct cecal microbiomes that resulted from exposure to antibiotics [@Schubert2015;@Jenior2017]. This is supported by the hypothesis that each differentially sensitized gut environment possesses an alternative nutrient niche landscape and *C. difficile* is able to adjust its metabolism accordingly. Although it is well established that *C. difficile* is able to colonize these communities, it is yet to be determined whether the differences in the metabolic capacity of communities following antibiotic treatment correlate with prolonged *C. difficile* colonization. Defining the functional status of the resident microbiota for any disease has been difficult and has led to a limited understanding of the specific species interactions that occur with *C. difficile* during infection. To address this fundamental question, we employed a murine model of CDI where the murine gut microbiota are disturbed with antibiotics and assessed for the ability of *C. difficile* to colonize and cause disease. We characterized the microbiome utilizing untargeted metabolomics and metagenome-enabled metatransciptomics. This approach allowed us to not only characterize the metabolic output of the community, but to also identify which subgroups of bacteria were differentially active under these conditions. Our data supports the hypothesis that *C. difficile* colonization indeed alters community-level gene expression and the metabolome of these communities. Furthermore, in mice that cleared the infection, despite shifts in species abundance, the metatrascriptome and metabolome changed very little. This work highlights the need for a better appreciation of the effect CDI has on the gut microbiota is needed to develop more successful targeted therapies that eliminate *C. difficile* after persistent colonization.


###Results

**Distinct antibiotic pretreatments are associated with different patterns of clearance following primary infection.**
Conventionally-reared SPF mice were pretreated with one of three different antibiotics to sensitize the animals to *C. difficile* colonization. The selected antibiotics were streptomycin, cefoperazone, and clindamycin (Table S1). Each drug was chosen because they represented different spectrum of activity, had the ability to to reduce *C. difficile* colonization resistance in a mouse model [@Schubert2015], had distinct and significant impacts on the structure and diversity of the cecal microbiota (all *p* < 0.001; Fig. S1A), and the infected communities differed in their to recover and clear the infection (Fig. 1B). Selection of the toxigenic *C. difficile* strain 630 was based on its moderate clinical severity in mouse models [@Theriot2011], previous studies of *in vitro* metabolism [@Bouillaut2015], and well-annotated genome [@VanEijk2015]. Briefly, mice were pretreated with the respective antibiotic and were subsequently challenged with ~1x10^3^ *C. difficile* str. 630 spores via oral gavage (Fig. 1A). We then monitored for disease over the following 10 days and the carriage of *C. difficile* in feces over time. In each antibiotic pretreatment model, we observed equally high levels of *C. difficile* colonization on the day after infection; however, over the following 8 days, only clindamycin-pretreated mice cleared the infection while the other two pretreatments remained highly colonized (Fig. 1B).


***C. difficile* differentially expresses virulence factors across separate antibiotic pretreatments.**
We hypothesized that the clindamycin-pretreated mice cleared the infection while animals in the other groups did not because the pretreated microbiota occupied niche space that with greater overlap with that of *C. difficile*. It had been previously demonstrated that *C. difficile* virulence factor expression is regulated by availability of certain nutrients in the environment [@Bouillaut2015], so we first sought to measure spore and toxin production. Continuing with the same pretreatment and primary infection protocols used in the 10-day colonization experiments, we chose to focus our analysis on 18-hours post-infection to the assess behavior of *C. difficile* directly prior to the beginning of clearance and utilized this time period for the remainder of experiments. This end point corresponded with a previous study where *C. difficile* reached maximum cecal vegetative colony forming units (cfu) with few detectable spores [@Koenigsknecht2015]. We also elected to examine cecal content because it was more likely to be a site of active bacterial metabolism compared to stool and would allow for an assessment of functional differences in the microbiota. At 18 hours after infection, there were no significant differences in the number of vegetative cells between any antibiotic-pretreatment group. All susceptible mice were colonized with ~1x10^8^ vegetative (cfu per gram of cecal content and untreated mice maintained *C. difficile* colonization resistance (Fig. 1C). We also measured both spore production and toxin activity as activation of both processes has been linked to environmental concentrations of specific growth nutrients [@Bouillaut2015]. Despite having similar vegetative *C. difficile* cfu, varying levels of both spore cfu and toxin titer were observed across each of the antibiotic pretreatments. These results showed that *C. difficile* colonized different antibiotic-pretreated mice to consistently high levels, and the distinct pretreatments corresponded with moderate differences in the level of *C. difficile* virulence factors.


**Initial *C. difficile* colonization levels were consistent in spite of significantly different community structures.**
It has not yet been established whether *C. difficile* colonization impacts the structure of the gut microbiota during infection in susceptible mice. To evaluate changes in bacterial population abundances in response to antibiotic perturbation and *C. difficile* colonization, we sequenced the V4 region of the 16S rRNA gene from the cecal content of both mock and *C. difficile*-infected mice across antibiotic pretreatment models. To focus our analysis specifically on the surrounding bacterial communities we removed all *C. difficile* 16S rRNA gene sequences, which represented less than 3.54% of sequencing reads in any infection group. We confirmed that each antibiotic pretreatment significantly impacted both cecal community structure (Bray-Curtis distances) and diversity (inverse-Simpson) compared to untreated control mice (all *p* < 0.001, Fig. S1A & S2B). We then confirmed that each antibiotic pretreatment induced distinct and significant shifts in the community structure (all *p* < 0.001, Fig. S1C). The composition of streptomycin-pretreated communities was more variable between cages, but was generally enriched for members of phylum *Bacteroidetes* (Fig. 2A). Cefoperazone and clindamycin-pretreated cecal communities were consistently dominated by members of the *Lactobacillaceae* and *Enterobacteriaceae* families, respectively (Fig. 2A). Despite variation in the community structures generated by streptomycin pretreatment, those communities were colonized by *C. difficile* to the same level (Fig. 1B, 1C, & S1).

Next, we measured the effect that *C. difficile* colonization had on overall community structure and composition within each of the antibiotic groups individually. We found that the structure of both streptomycin and clindamycin-pretreated infected communities were significantly different from their corresponding uninfected controls (streptomycin *p* = 0.014, Fig. S1D; clindamycin *p* = 0.003, Fig. S1F). Cefoperazone-pretreated communities did not significantly differ from their control group (Fig. S1E). We then sought to identify specific OTUs that significantly differed in abundance between mock and *C. difficile*-infected communities within each pretreatment group. Utilizing LEfSe differential abundant feature detection [@Segata2011], we identified 16 OTUs that discriminated between infected and uninfected communities in at least one antibiotic pretreatment (Fig. 2B-2D); however, these OTUs were generally near the limit of detection and very few were present across multiple groups tested. This suggested that CDI susceptible communities were only modestly impacted by the presence of *C. difficile*.


**Distinct antibiotic classes lead to alternative markers of *C. difficile* colonization susceptibility.**
Pretreatment with antibiotics not only alters the structure of the resident microbiota, but also has a dramatic impact on the intestinal metabolome [@Antunes2011; @Jump2014; @Theriot2014]. To test the metabolic features of the altered communities, we performed untargeted metabolomic analysis on the cecal contents that were also utilized in the 16S rRNA gene sequencing. A total of 727 distinct metabolites were identified. First, we characterized the differences between the metabolomes of the mock-infected communities to measure the impact of antibiotic pretreatment that generated *C. difficile*-susceptible conditions (Fig. 3). All of the antibiotic pretreatments significantly altered the cecal metabolome compared to untreated, *C. difficile*-resistant mice as quantified by Bray-Curtis dissimilarity (*p* < 0.001; Fig. 3A). As with the differences between resistant and susceptible states, the patterns of specific metabolite concentrations diverged depending on the antibiotic pretreatment (Fig. S3 & S5). These results demonstrated that each pretreatment lead to distinct susceptible metabolomic environments.

We were interested in identifying those metabolites with the highest degree of change between resistant and susceptible groups. Due to the large number of individual metabolites with significant differences in relative concentration between groups, we employed the Random Forest machine-learning algorithm [@Liaw2002] to highlight those metabolites that most easily differentiated the antibiotic groups in the absence of infection. To further limit the analysis to only the most informative metabolite, we ranked metabolites in order of highest mean decrease accuracy (i.e. the amount in which their removal negatively impacts correct sample classification) and reported the top 7 metabolites for each resistant/susceptible comparison (Fig. 3C). These lists primarily included precursors or intermediates of polysaccharide fermentation such as chiro\−inositol, malonate, erythritol, 4\−guanidinobutanoate, and soyasaponin II, the majority of which were increased in susceptible conditions. Decreases in these metabolites *in vivo* have previously been associated with a reduction in the normal levels of polysaccharide fermentation present in the intact microbiota [@Jump2014]. N\−methylpipecolate was the only metabolite we identified that consistently decreased during antibiotic pretreatment. This molecule is a by-product of amino acid catabolism and is typically oxidized under normal conditions back into glycine [@Goyer2004]. These data support the hypothesis that amino acid catabolism is disrupted at a community-level following antibiotic pretreatment and may suggest open nutrient niches for *C. difficile*, a known fermenter of peptides [@Fonknechten2010]. We then applied this technique to identify metabolites that distinguish antibiotic pretreatment groups from each other (Fig. 3D). Interestingly, several additional carbohydrate or amino acid catabolism by-products were highlighted by these means including hydroxyisocaproate, methylvalerate, glutamylmethionine, and N\−carbamoylaspartate [@Jump2014]. Additionally, sucrose is a *C. difficile* growth substrate [@Nakamura1982] which further promoted the hypothesis of differentially vacated nutrient niches due to antibiotic pretreatment that are now accessible to *C. difficile* (Fig. 2D). These data suggested that the populations responsible for carbohydrate fermentation that result in short-chain fatty acid (SCFA) production may be differentially affected by different classes of antibiotics. The distinct antibiotics allowed for different forms of nutrient catabolism to survive the pretreatment and potentially implies that there are alternative metabolite profiles that impact *C. difficile*'s ability to compete with the indigenous microbiota.

Following this unsupervised approach for identification of susceptibility markers, we also measured the differences in relative concentration of metabolites previously connected to potential *C. difficile* colonization susceptibility [@Theriot2014]. We assessed differences in bile acids, which are deeply connected to the life cycle of *C. difficile* [@Sorg2010] and whose bioconversion by the microbiota has been implicated as a driver of colonization resistance [@Buffie2014]. We found that there were no persistently increased bile acid across the chosen antibiotic pretreatments (Fig. S2a). There were also no amino acids that were always highly available (Fig. S2B). This suggested that despite varying efficiencies, *C. difficile* can recognize a subset of these molecules in various contexts leading to germination and outgrowth. In contrast, we found that several carbohydrates were significantly increased across all antibiotic groups including arabitol/xylitol, ribitol, and sucrose (Fig. S2B). Together, our results supported that each susceptible environment was distinguishable from the others, with its own subset of enriched *C. difficile* growth substrates. This could be an indication that particular competitors were eliminated during antibiotic pretreatment, or those community members normally responsible for the consumption of these metabolites have altered their metabolic program to exploit alternative nutrient sources.


***C. difficile* colonization alters each susceptible cecal metabolome distinctly.**
Following the changes to the cecal metabolome in response to antibiotic pretreatment, we assessed the degree to which *C. difficile* colonization altered the cecal metabolome of susceptible animals. We hypothesized that the introduction of a new competitor, *C. difficile*, would impact the metabolome either through signatures of its own metabolism or by causing a shift in the metabolism of other members in the surrounding community. First, in a similar approach to identifying susceptibility markers, we observed the global Bray-Curtis dissimilarities of mock-infected and *C. difficile*-infected metabolomes within each antibiotic pretreatment group individually. We compared the cecal metabolomes from all mice across pretreatment groups, and were unable to detect a consistent difference between groups (*p* = 0.075; Fig. S3A). We moved on to individual antibiotic groups and found that both streptomycin (*p* = 0.039) and cefoperazone (*p* = 0.016) pretreated metabolomes deviated significantly from that of mock infection (Fig. 4A-4B). However, clindamycin-pretreated cecal metabolomes were not significantly altered by the presence of the pathogen (*p* = 0.127; Fig. 4C). These results diverged from what was seen in the paired OTU relative abundance results where instead the cecal community structure of mice pretreated with cefoperazone was unchanged and clindamycin-pretreated communities were significantly different (*p* = 0.003; Fig. 4B, 4C). Interestingly, streptomycin-pretreated microbiomes were significantly altered by *C. difficile* infection at both the OTU and metabolomic levels (Fig. 4A & S1D). These data indicated that large shifts of populations in the cecal microbiota were not implicitly associated with concordant shifts in the metabolome. This supported the hypothesis that divergent community structures can ultimately share a convergent metabolic output despite changes to community structure and membership.

We then sought to identify changes in those metabolites that were potential markers of *C. difficile* infection through a pooled analysis across all models tested. To accomplish this, we again applied the Random Forest machine-learning approach. We were able to distinguish those microbiomes infected with *C. difficile* from those that were not and identified the 10 metabolites with the greatest MDA (Fig. S3B). The strongest single predictor and only metabolite among the top 10 that was increased during infection was 5-aminovalerate. Presence of 5-aminovalerate appears to be dependent on bacterial processes as relative concentrations of the molecule were significantly lower in ceca of antibiotic-pretreated, mock-infected animals (Fig. 4G). Conversely, it was found to be highly abundant in most infected animals compared to both susceptible and resistant controls. Only in clindamycin-pretreatment, the only group to reduce *C. difficile* colonization over 10 days, did the concentration of 5-aminovalerate remain near undetectable in a subset of samples. This amino acid analog is a known byproduct of D-proline fermentation in *C. difficile* [@Neumann-Schaal2015]. *C. difficile* is able to catabolize proline along with glycine through a set of paired biochemical reactions known as Stickland fermentation [@Bouillaut2013]. Along these lines, when we measured for unified metabolite signals across all antibiotic pretreatment models, proline and 3 proline-conjugated amino acids (i.e. pro-hydroxy-pro, thioproline, and trans-4-hydroxyproline) were significantly decreased during infection (Fig. S3B; all *p* < 0.001). These combined results demonstrated that while distinct metabolic challenges may exist in each susceptible metabolome, Stickland fermentation could be a preferred energy acquisition pathway for *C. difficile* *in vivo*. Random Forest machine-learning models were able to reliably classify infected and uninfected mice in both streptomycin and cefoperazone pretreatments (Fig. 4A-4B, Out-of-bag error = 0%). However, infection did not provide for the same level of discrimination in clindamycin-pretreated mice (Fig. 4C, Out-of-bag error = 44.44%). None of the top 5 MDA metabolites in each antibiotic pretreatment were shared across all groups during individual analysis. These data supported our hypothesis that not only does each antibiotic pretreatment create different community structures, but that the metabolism of these distinct communities responds to *C. difficile* colonization in a unique manner. However, it was not possible to isolate where changes to the metabolome were a result of altered community metabolism, altered host metabolism, or from *C. difficile*.


***C. difficile* colonization induced shifts in the expression of several catabolic pathways and nutrient acquisition systems across susceptible communities.**
Combined, 16S rRNA gene sequencing and metabolomics results demonstrated that antibiotic pretreatments resulted in distinct bacterial communities that led to altered metabolite profiles. It is also possible that metabolism of the host or *C. difficile* contributed to the differences in the metabolomes during infection. To gain a more specific understanding of how the microbiota or *C. difficile* shaped the metabolic environment, we employed a metagenome-enabled metatranscriptomic shotgun sequencing approach with paired DNA and RNA samples collected from the cecal content of the mice used in the previous analyses. Metagenomic reads from mock-infected cecal communities were assembled *de novo* into contigs and putative genes were identified resulting in 234,124 (streptomycin), 83,096 (cefoperazone), and 35,977 (clindamycin) potential genes in each metagenome. Streptomycin pretreatment resulted in a significantly more diverse community (Fig. S1A) than other groups, so a more diverse metagenome was expected. Utilizing metatranscriptomic read abundances normalized to corresponding metagenomic coverage per gene, we first focused on differences in gene expression in broadly defined pathways and gene categories for each antibiotic pretreatment with and without *C. difficile*-infection. Based on the metabolomic results, we hypothesized that pathways with the greatest differences would include those involved in the metabolism of carbohydrates and amino acids. We then calculated the difference in cDNA abundance for each pathway between infected and uninfected conditions, represented as delta-cDNA abundance. To highlight the largest differences, we limited the analysis to the top 5 KEGG pathways with the most change between mock and *C. difficile*-infected conditions within each pretreatment (Fig. 5A). In streptomycin and clindamycin-pretreated mice, increased expression of KEGG pathways was observed in the *C. difficile*-infected metatranscriptomes. Both groups displayed large changes in amino sugar metabolism and ABC transporters, however other distinct carbon metabolism pathways were upregulated. Genes for glycolysis and oxidative phosphorylation were overrepresented in streptomycin pretreated mice while starch/sucrose metabolism and PTS systems were more abundantly associated with clindamycin-pretreated mice. Together these shifts suggested that communities differentially adapt carbon metabolism pathways in response to colonization of *C. difficile*. Conversely in cefoperazone-pretreated mice, a large number of genes were instead over-expressed in the absence of *C. difficile*. These pathways included the replication or manipulation of genetic material (i.e. RNA Processing, tRNA Synthesis, & Homologous Recombination). Among the genes that were upregulated in the presence of *C. difficile*, there were many genes involved in anaerobic glucose metabolism and select ABC-transporters (Table S3). These results indicate that the cecal microbiota of infected mice shifted its metabolism toward catabolizing simple carbohydrates.

We then performed a more fine-scale analysis of changes at the gene level, by selecting the genes in each antibiotic pretreatment group with the largest disparity in normalized cDNA abundance between mock and *C. difficile*-infected groups (Fig. 5B-5D). In agreement with pathway-level differences in expression, the majority of genes with the greatest difference between mock and infected mice belonged to the pathways highlighted in Fig. 5A with three additional pathways relating to amino acid metabolism (I: Glutathione metabolism, J: Valine/Leucine/Isoleucine metabolism, & K: Glycine/Serine/Threonine metabolism). Numerous genes for transport of simple carbohydrates and glycolysis were also differentially overrepresented in infected conditions across pretreatment groups when compared to mock infection of untreated, resistant communities (Fig. S4). These data expand our interpretation of pathway-level analysis and support our hypothesis that *C. difficile* colonization leads to changes in community-level expression of genes for nutrient acquisition and catabolism. 


***C. difficile* colonization corresponds with large-scale changes in expression of genes from specific bacterial taxa.**
As not all bacterial taxa share identical metabolic capabilities, we hypothesized that expression in specific subsets of bacteria were differentially affected by the presence or metabolic activity of *C. difficile*. We sought to delineate the transcriptomic contributions of separate bacterial taxa within each metatranscriptomic dataset. To accomplish this we utilized the genus level taxonomic information associated with each KEGG annotation to identify which group likely contributed a given gene to the metagenome. Many genes in the KEGG database are annotated as hypothetical or uncharacterized but still possess a taxonomic annotation. This resulted in substantially more genes from the total being conserved for analysis in each group. With these data, we narrowed the focus to transcription for genera that represented >0.01% of genes receiving taxonomic annotations in any of the metagenomic assemblies. We then directly compared the normalized cDNA abundances for each gene between infected and uninfected states for each antibiotic pretreatment and applied linear correlation to identify distinct patterns of expression (Fig. 3A-3C). This resulted in 2473 outliers at an average distance of 2.545 arbitrary units (AU) associated with streptomycin, 2930 outliers at an average distance 3.854 AU with cefoperazone, and only 727 outliers at an average distance of 2.414 AU in clindamycin pretreatment. Overall, the clindamycin pretreatment was associated with the fewest expression outliers between uninfected and infection conditions compared with those of the other antibiotic groups. This indicated that the community gene expression was not significantly altered in the presence of *C. difficile*.

As many genes lack a specific functional annotation in KEGG but do map to individual bacterial species, we continued the analysis by measuring change in transcript abundances contributed by the more prevalent genera. Incorporation of the genus-level taxonomic information for each transcript revealed that outlier genes were contributed by underrepresented genera. In streptomycin-pretreated mice, the most prominent differences were in 937 genes belonging to *Lactobacillus* that were upregulated with *C. difficile* infection (Fig. 6A). In cefoperazone-pretreated mice 2290 genes belonging to *Bacteroides* were upregulated in mock infected mice (Fig. 6B). A consistent trend in streptomycin and cefoperazone pretreatments was an overrepresentation of highly expressed genes from genera belonging to *Bacteroidetes* during mock infection. The metatransciptomes within both of these pretreatment conditions poorly correlated between mock and infected conditions, indicating a high degree of change induced by *C. difficile* colonization (*r* = 0.03 & *r* = 0.03). In clindamycin-pretreated mice the largest difference in transcription was for 510 *Lactobacillus* genes that were upregulated during CDI (Fig. 6C). Infected and uninfected metatranscriptomes associated with this antibiotic more strongly correlated than either other pretreatment (*r* = 0.86), further supporting that *C. difficile* colonization had a low impact on transcription of the cecal microbiota. These data indicate that *C. difficile* may differentially modify the transcriptional activity of separate microbial populations based on the context of the community in which it colonizes.


###Discussion

Our results demonstrate that distinct intestinal ecosystems are differently impacted by *C. difficile* colonization and that these changes to community metabolism could have implications on the ability of the pathogen to persist in those environments. Furthermore, discordant modifications to OTU abundances, transcriptional activity of the microbiota, and cecal metabolome shortly after infection suggest that *C. difficile* manipulates the niche landscape of the intestinal tract in order to better maintain long-term colonization. This hypothesis is best supported by the disparity of community-level phenotypes between the clindamycin and cefoperazone/ or streptomycin pretreatment models. Only in the clindamyin pretreatment model were animals able to go on to clear *C. difficile*. Unlike the other conditions, clindamycin-pretreated communities were significantly altered in community structure following *C. difficile* colonization, but not at the metabolomic level (Fig. S1F & 4A). This disconnect between community structure and metabolic output was explained by their similarity in gene expression (Fig. 4D & 5C). Collectively, these results advance the idea that for *C. difficile* to maintain colonization it must exploit and restructure the available niche space. 

Instances of active nutrient niche restructuring in the gut have been documented previously for prominent symbiotic bacterial species in gnotobiotic mice [@Mahowald2009], but never before in a conventionally-reared animal model of infection following antibiotic pretreatment. Interestingly, the taxonomic groups that were represented as outliers in the normalized metatranscriptomes of each antibiotic pretreatments were rare in their cecal community (on average <5% of community; Fig. 2A & 6). Previous studies have found that rare taxonomic groups, even those at a low abundance as a result of a spontaneous perturbation, may have disproportion effects on the metabolic environment of the community at large [@Jousset2017]. For example, this strategy has been observed in temperate lakes where conditionally rare microbes were found to be far more metabolically active than highly abundant taxa [@Shade2014], and this concept would likely apply to bacterial groups recovering population density following cessation of antibiotic pretreatment. As such, *C. difficile* may preferentially seek to compete with these organisms to ultimately affect greater change to the entire ecosystem and open a long-lasting nutrient niche. While this hypothesis requires further exploration, it provides an ecological framework to study the interactions between *C. difficile* and members of susceptible communities.

This study is one of the first *in vivo* observations that a medically relevant bacterial pathogen may alter the metabolic activity of a host-associated community to promote its own colonization. Others have identified potential metabolite markers of *C. difficile* infection in patient feces [@Rojo2015], but they were not able to make correlations with changes in community metabolism that were afforded to us by paired untargeted metabolomic analyasis and metatranscriptomic sequencing. In a recent study, a tick-vectored bacterial pathogen altered the ability of the resident microbiota of the tick by interrupting proper biofilm formation and allowing lasting colonization [@Abraham2017]. Furthermore, it was recently found that bacterial metabolic generalists may be more likely to actively antagonize the growth of other species in an environment which they are colonizing [@Russel2017]. In previous work from our group, we found that *C. difficile* has a wide nutrient niche-space *in vivo* and is most likely utilizes metabolic generalism in order to colonize diverse gut microbiomes [@Jenior2017]. The ability to simultaneously antagonize the metabolism of surrounding species in cecal environments that support persistence would explain the more significant shifts in metatranscriptomic expression. While we acknowledge that this study may not elucidate the specific mechanism by which this interaction occurs, the combined systems analysis strengthens each individual level of observation and only when employed together does a clearer definition of *C. difficile*-related microbial ecology in the gut emerge. This research lays the groundwork for a more rationale consideration of the metabolic functionalities of bacterial taxa to consider when attempting to rebuild *C. difficile* colonization resistance across differentially perturbed gut environments.

In spite of consistent signals across multiple levels of -omics datasets, possible shortcomings to our interpretation of the presented data do exist. First, as with all transcriptomic studies, the relative level of mRNA detected for a given gene does not necessarily reflect the amount of functional protein made by a cell. Furthermore, interpretation of timing may also be an issue since a large influx of transcript for a specific product may signal an initial upregulation before subsequent translation has been able to occur. This also omits consideration of any post-translational modifications that are required for ultimately functional enzymes. Additionally, due to the low relative abundance of *C. difficile* in these communities as well as the high density of other surrounding bacterial species required focusing sequencing efforts on depth rather than multiple replicates within each group. Greater transcript read abundance per gene allows for improved survey for the activity of lowly abundant species as well as greater confidence in genes found to be highly transcribed. Although the lack of animal-based replication for the metatranscriptomic data does potentially limit the ability to generalize our results, this approach has been successfully utilized by numerous groups in the past to accurately characterize transcriptionally activity across communities of bacteria [@Sheik2014;@Franzosa2014;@Jorth2014;@Jenior2017]. In terms of metabolomics, alternative possible interpretations of the data also exist. For example, we considered metabolites that did not change in concentration between uninfected and infected conditions as not impacted by changes in bacterial metabolism induced by *C. difficile* colonization. However, this may instead indicate that the metabolism of *C. difficile* itself may simply replace the level present in the uninfected community. Such instances may not be detectable through untargeted mass spectrometry alone, however the combination of methods utilized here present a much more unified description of the system than any of the component techniques alone.

Several groups have attempted to identify single bacterial species or limited strain consortia that are able to replicate this effect, but each has been met with incomplete restoration of colonization resistance or function [@Reeves2012;@Lawley2012;@Petrof2013;@Buffie2014]. The effect of *C. difficile* colonization on community metabolic activity could also be linked to pathogen strain and may offer an explanation to the propensity of some strains to persist over others [@Adlerberth2014]. Moreover, the current work contributes to the existing concept that the healthy gut microbiota maintains colonization resistance to *C. difficile* by outcompeting the pathogen for preferred nutrient niche space. Ultimately, our results suggest that each susceptible and subsequently infected microbiome may be unique and require specific microbes or functionalities to restore colonization resistance to *C. difficile* in that specific context. Conversely, colonization resistance against *C. difficile* may be the result of contributions by distinct subcommunities of bacteria across each unique resistant gut community. As the microbiome is so intimately connected to colonization resistance against the bacterium, it has become imperative to understand what factors allow some gut environments to be persistently colonized while others are not. This research lays the groundwork for future studies to assess context dependent restoration of *C. difficile* colonization resistance and what factors are able to interfere with the ability of *C. difficile* to modify gut ecology in order to promote clearance.


###Materials and Methods

**Animal care and antibiotic administration.** For a more detailed description of the animal model, refer to Jenior et al. [@Jenior2017]. Briefly, approximately equal numbers of male and female conventionally-reared six-to-eight week-old C57BL/6 mice were randomly assigned to each experimental group. They were administered one of three antibiotics; cefoperazone, streptomycin, or clindamycin before oral *C. difficile* infection (Table S1). All animal protocols were approved by the University Committee on Use and Care of Animals at the University of Michigan and carried out in accordance with the approved guidelines.

***C. difficile* infection and necropsy.** On the day of challenge, 1x10^3^ *C. difficile* spores were administered to mice via oral gavage in phosphate-buffered saline (PBS) vehicle. Mock-infected animals were given an oral gavage of 100 ul PBS at the same time as those mice administered *C. difficile* spores. 18 hours following infection, mice were euthanized by CO~2~ asphyxiation and necropsied to obtain the cecal contents. Aliquots were immediately flash frozen for later DNA extraction and toxin titer analysis. A third aliquot was transferred to an anaerobic chamber for quantification of *C. difficile* abundance. The remaining content in the ceca was mixed in a stainless steel mortar housed in a dry ice and ethanol bath. Cecal contents from all mice within each pretreatment group were pooled into the mortar prior to grinding to a fine powder. The ground content was then stored at -80&deg;C for subsequent RNA extraction. For 10-day colonization studies, fresh stool was collected from infected mice each day beginning on the day of infection. Mice were monitored for overt signs of disease and were euthanized after the final stool collection.

***C. difficile* cultivation and quantification.** Cecal samples were weighed and serially diluted under anaerobic conditions with anaerobic PBS. Differential plating was performed to quantify both *C. difficile* spores and vegetative cells by plating diluted samples on CCFAE plates (fructose agar plus cycloserine, cefoxitin, and erythromycin) at 37&deg;C for 24 hours under anaerobic conditions [@Wilson1982]. In parallel, undiluted samples were heated at 60&deg;C for 30 minutes to eliminate vegetative cells and enrich for spores [@Sorg2010]. These samples were serially diluted under anaerobic conditions in anaerobic PBS and plated on CCFAE with taurocholate at 37&deg;C for 24 hours. Plating was simultaneously done for heated samples on CCFAE to ensure all vegetative cells had been eliminated. Quantification of *C. difficile* cfu for the 10-day colonization experiments was performed from stool using TCCFAE to measure total *C. difficile* load in these animals over time.

***C. difficile* toxin titer assay.** To quantify the titer of toxin in the cecum a Vero cell rounding assay was performed as described elsewhere [@Leslie2015;@Jenior2017]. Briefly, filtered-sterilized cecal content was serially diluted in PBS and added to Vero cells in a 96-well plate where it was then incubated at 37&deg;C for 24 hours. Plates were then inspected for evidence of cell rounding.

**DNA/RNA extraction and sequencing library preparation.** DNA for shotgun metagenomic and 16S rRNA gene sequencing was extracted from approximately 50 mg of cecal content from each mouse using the PowerSoil-htp 96 Well Soil DNA isolation kit (MO BIO Laboratories) and an epMotion 5075 automated pipetting system (Eppendorf). The V4 region of the bacterial 16S rRNA gene was amplified using custom barcoded primers [@Kozich2013]. Equal molar ratios of raw isolated DNA within each treatment group were then pooled and ~2.5 ng of material was used to generate shotgun libraries with a modified 10-cycle Nextera XT genomic library construction protocol (Illumina). This was done to mimic the pooling strategy necessary for metatranscriptomic library preparation. Final libraries were pooled at equal molar ratios and stored at -20&deg;C. For RNA extraction, a more detailed description of the procedure can be found in [@Jenior2017]. Briefly, immediately before RNA extraction, 3 ml of lysis buffer (2% SDS, 16 mM EDTA and 200 mM NaCl) contained in a 50 ml polypropylene conical tube was heated for 5 minutes in a boiling water bath [@Lopez-Medina2011]. The hot lysis buffer was added to the frozen and ground cecal content. The mixture was boiled with periodic vortexing for another 5 minutes. After boiling, an equal volume of 37&deg;C acid phenol/chloroform was added to the cecal content lysate and incubated at 37&deg;C for 10 minutes with periodic vortexing. The mixture was the centrifuged at 2,500 x g at 4&deg;C for 15 minutes. The aqueous phase was then transferred to a sterile tube and an equal volume of acid phenol/chloroform was added. This mixture was vortexed and centrifuged at 2,500 x g at 4&deg;C for 5 minutes. The process was repeated until aqueous phase was clear. The last extraction was performed with chloroform/isoamyl alcohol to remove acid phenol. An equal volume of isopropanol was added and the extracted nucleic acid was incubated overnight at -20&deg;C. The following day the sample was centrifuged at 12000 x g at 4&deg;C for 45 minutes. The pellet was washed with 0&deg;C 100% ethanol and resuspended in 200 ul of RNase-free water. Following the manufacturer's protocol, samples were then treated with 2 ul of Turbo DNase for 30 minutes at 37&deg;C. RNA samples were retrieved using the Zymo Quick-RNA MiniPrep according the manufacturer's protocol. The Ribo-Zero Gold, immediately before RNA extraction, 3 ml of lysis buffer (2% SDS, 16 mM EDTA and 200 mM NaCl) contained in a 50 ml polypropylene conical tube was heated for 5 minutes. RNA Removal Kit Epidemiology was then used to deplete prokaryotic and eukaryotic rRNA from the samples according the manufacturer's protocol. Stranded RNA-Seq libraries were made constructed with the TruSeq Total RNA Library Preparation Kit v2, both using the manufacturer's protocol. Completed libraries were pooled in equal molar ratios within their respective groups and stored at -20&deg;C until time of sequencing.

**High-throughput sequencing and raw read curation.**
Sequencing of 16S rRNA gene amplicon libraries was performed using an Illumina MiSeq sequencer as described previously [@Kozich2013]. The 16S rRNA gene sequences were curated using the mothur software package (v1.36) as described in [@Jenior2017]. Shotgun metagenomic sequencing was performed in 2 phases. Libraries from mock-infected communities, that were also to be utilized for *de novo* contig assembly, were sequenced using an Illumina HiSeq 2500 on 2x250 paired-end settings and was repeated across 2 lanes to normalize for inter-run variation. *C. difficile*-infected metagenomic libraries were sequenced with an Illumina NextSeq 300 with 2x150 settings across 2 runs to also normalize for inter-run variation. These efforts resulted in an average of 280 million paired raw reads per sample. Metatranscriptomic sequencing was performed on an Illumina HiSeq 2500 with 2x50 settings and was repeated across 4 lanes for normalization and to obtain necessary coverage [@Franzosa2014]. This gave an average of 380 million raw cDNA reads per library. Both metagenomic and metatranscriptomic sequencing was performed at the University of Michigan Sequencing Core. Raw sequence read curation for both metagenomic and metatranscriptomic datasets was performed in a two step process. Residual 5-prime and 3-prime Illumina adapter sequences were trimmed using CutAdapt [@Martin2011] on a per library basis. Reads were quality trimmed using Sickle [@Joshi2011] with a quality cutoff of Q30. This resulted in approximately 270 million reads per library (both paired and orphaned) for both metagenomic and metatranscriptomic sequencing. Actual read abundances for individual metagenomic and metatranscriptomic sequencing efforts can be found in Table S2.

**Metagenomic contig assembly and gene annotation.** 
Metagenomic contigs were assembled using Megahit [@Li2014] with the following settings; minimum kmer size of 87, maximum kmer size of 127, and a kmer step size of 10. Prodigal was utilized to to identify putative gene sequences, and were screened for a minimum length of 250 nucleotides. These sequences were translated to amino acids and peptides were annotated against the KEGG protein database [@Ogata1999] using Diamond implementation of BLASTp [@Buchfink2015]. Peptide-level gene annotations were assigned to the corresponding nucleotide sequence, and genes failing to find a match in KEGG were preserved as unannotated genes. Final nucleotide fasta files with KEGG annotations were then utilized in the construction of Bowtie2 [@Langmead2012] mapping databases from downstream analyses.

**DNA/cDNA read mapping and normalization.**
Mapping was accomplished using Bowtie2 [@Langmead2012] and the default stringent settings. Optical and PCR duplicates were then removed using Picard MarkDuplicates (http://roadinstitute.github.io/picard/). The remaining mappings were converted to idxstats format using Samtools [@Li2009] and the read counts per gene were tabulated. Discordant pair mappings were discarded and counts were then normalized to read length and gene length to give a per base report of gene coverage. Transcript abundance was then normalized to gene abundance to yield overall level of upregulation for each gene. Unless indicated otherwise, each collection of reads was then 1000-fold iteratively subsampled to 90% of the lowest sequence total within each analysis, and a median expression value for each gene was calculated.

**Quantification of *in vivo* metabolite relative concentrations.** 
Metabolomic analysis was performed by Metabolon (Durham, NC), or a detailed description of the procedure refer to [@Jenior2017]. Briefly, all methods utilized a Waters ACQUITY ultra-performance liquid chromatography (UPLC) and a Thermo Scientific Q-Exactive high resolution/accurate mass spectrometer interfaced with a heated electrospray ionization (HESI-II) source and Orbitrap mass analyzer at 35,000 mass resolution. Samples were dried then reconstituted in solvents compatible to each of the four methods. The first, in acidic positive conditions using a C18 column (Waters UPLC BEH C18-2.1x100 mm, 1.7 um) using water and methanol, containing 0.05% perfluoropentanoic acid (PFPA) and 0.1% formic acid (FA). The second method was identical to the first but was chromatographically optimized for more hydrophobic compounds. The third approach utilized a basic negative ion optimized conditions using a separate dedicated C18 column. Basic extracts were gradient eluted from the column using methanol and water, however with 6.5mM Ammonium Bicarbonate at pH 8. Samples were then analyzed via negative ionization following elution from a hydrophilic interaction chromatography column (Waters UPLC BEH Amide 2.1x150 mm, 1.7 um) using a gradient consisting of water and acetonitrile with 10mM Ammonium Formate, pH 10.8. The MS analysis alternated between MS and data-dependent MS n scans using dynamic exclusion. The scan range varied slighted between methods but covered 70-1000 m/z. Library matches for each compound were checked for each sample and corrected if necessary. Peaks were quantified using area under the curve.

**Statistical methods.** All statistical analyses were performed using R (v.3.2.0). Significant differences between community structure of treatment groups from 16S rRNA gene sequencing were determined with AMOVA in the mothur software package. Significant differences of inverse Simpson diversity, cfu, toxin titer, and metabolite concentrations were determined by Wilcoxon signed-rank test with Benjamini-Hochberg correction using a study-wide Type I error rate of 0.05. Undetectable points used half the limit of detection for cfu and toxin statistical calculations. LEfSe analysis with OTU data was performed with a cutoff that each OTU must appear in all samples from their respective groups to be considered true signal (n=9). Random forest was performed using the implementation in R [@Breiman2001], with the informative threshold of MDAs greater than the absolute value of the lowest MDA defined by [@Segal2004]. Distances of outlier points from center line during metatranscriptomic comparisons was accomplished using 2-dimensional linear geometry.


###Funding Information

This work was supported by funding from the National Institutes of Health to PDS (R01GM099514, P30DK034933, U19AI09087, and U01AI124255), VBY (P30DK034933, U19AI09087, and U01AI124255), a Translational Research Education Certificate grant to JLL (MICHR; UL1TR000433), and was partially supported by a predoctoral fellowship from the Cellular Biotechnology Training Program to MLJ (T32GM008353).


###Acknowledgments

The authors would like to acknowledge Charles Koumpouras for assistance with DNA extractions and metabolomic sample preparation. We would also like to acknowledge members of the University of Michigan Germfree Mouse Center, University of Michigan Sequencing Core, and Metabolon for their assistance in experimental design, execution, and data collection. Pooled and quality trimmed transcriptomic and 16S rRNA gene amplicon read data from infection experiments are available through the NCBI Sequence Read Archive (SRA; PRJNA354635). Metagenomeic reads, mock-infected metatranscriptomic reads can be found also on the SRA (PRJNA415307). Data processing steps for beginning from raw sequence data to the final manuscript are hosted at https://github.com/mjenior/Jenior_Metatranscriptomics_eLife_2017.


**Author Contributions**
M.L.J. conceived, designed and performed experiments, analyzed data, and drafted the manuscript. J.L.L. performed experiments and contributed to the manuscript. V.B.Y conceived of experiments and contributed to the manuscript. P.D.S. interpreted data and contributed to the manuscript.

**Competing Interests**
The authors declare no conflicts of interest.


###Figure Legends

**Figure 1 | Experimental models of *C. difficile* infection and distinct virulence patterns.** **(A)** Experimental timelines of antibiotic pretreatment and infection mouse models. **(B)** *C. difficile* 630 CFU in stool of infected mice following each antibiotic pretreatment regimen over 10 days of infection. Median and interquartile range are shown for each time point. **(C)** Quantification of *C. difficile* cfu and toxin titer in cecal content across antibiotic treatment models after 18 hours of infection. Black lines indicate median values. Gray asterisks indicate significant difference from no antibiotic controls by Wilcoxon rank-sum test with the Benjamini-Hochberg correction (all *p* < 0.001). 

**Figure 2 | Impact of *C. difficile* colonization on other bacterial populations abundances in the gut microbiota.** **(A)** Relative abundance of family-level taxonomic classification for OTUs in each pretreatment group. **(B-D)** Discriminating OTUs with LEfSe analysis between Mock and *C. difficile*-infected communities within each pretreatment group, *p*-values from LEfSe are shown on the left with taxonomic information (all *p* < 0.05). Relative abundance of the respective OTUs from each mouse along with medians are shown for each pretreatment group. 

**Figure 3 | Metabolite markers of *C. difficile* infection susceptibility.** Results from UPLC/MS metabolomic analysis of cecal content from mice also used in previous analyses. Only mock-infected metabolomic results were included this analysis to identify markers of susceptibility. **(A-B)** NMDS ordinations of Bray-Curtis distances comparing the cecal metabolome dissimilarity of mice receiving no pretreatment or one of the three distinct classes of antibiotics. (A) Metabolomes of resistant mice are significantly different from antibiotic pretreated animals (*p* < 0.001). (B) Antibiotic pretreated metabolomes and also significantly distinct from one another (*p* < 0.001). Significant differences for NMDS analyses were calculated with AMOVA. **(C-D)** Scaled intensities of metabolites with highest mean decrease in accuracy (MDA) from random forest feature selection discriminating groups from (A-B). MDA is labeled in brackets beside each metabolite name and out-of-bag error from internal cross-validation is labeling along the bottom axis. Asterisks along the right axis indicate significant difference by Wilcoxon rank-sum test. (C) Relative concentrations of metabolites that distinguish cecal content from mice resistant to *C. difficile* colonization from susceptible animals. (D) Metabolites concentrations for those molecules that are able to differentiate antibiotic pretreatment groups. Multiple comparisons were accounted for using the Benjamini-Hochberg correction.

**Figure 4 |  *C. difficile* infection has differential effects on the cecal metabolome.** **(A-C)** NMDS ordinations of Bray-Curtis distances comparing infection groups within each antibiotic pretreatment; (A) Streptomycin, (B) Cefoperazone, and (C) Clindamycin. Significant differences were determined by AMOVA. **(D-F)** Relative concentrations of metabolites with highest Mean Decrease Accuracy (MDA) from Random Forest feature selection to discriminate between Mock and *C. difficile*-infected conditions within each antibiotic pretreatment. Respective MDA is labeled in brackets next to each metabolite name and out of bag error from internal cross-validation of each model is labeled under the plotting area. (D) Streptomycin pretreatment, (E) Cefoperazone pretreatment, and (F) Clindamycin pretreatment. **(G)** Relative concentrations of 5-aminovalerate across all experimental and control groups. Significant differences are labeled along the right axis and were calculated with Wilcoxon rank-sum test with Benjamini-Hochberg correction.

**Figure 5 | *C. difficile* infection alters community-level select pathway and gene expression of the gut microbiota across perturbed communities** Metatranscriptomic (cDNA) read abundances associated with each gene were normalized to their associated metaganomic (DNA) coverage, resulting in values that reflect upregulation. **(A)** Differences in read abundance for top 5 KEGG pathways with the largest amount of change between Mock and *C. difficile*-infected states within each antibiotic pretreatment. Values above the center line were expressed more during *C. difficile* infection, and those below the line were expressed more in Mock-infected animals. Dotted lines indicate average expression of pathways associated with each condition. **(B-D)** cDNA read abundances of the top 10 genes with the largest differences in expression within each indicated antibiotic pretreatment group. Shown are the expression levels for the genes displayed on the left during Mock (white) or *C. difficile* (black) infection. Gene names and member pathways indicated by letter codes along the left axis (pathway legend below). 

**Figure 6 | *C. difficile* colonization alters gene expression of taxonomic groups differentially between antibiotic pretreatments.** Each point represents a unique gene from the respective metagenomic assembly. Coordinates were determined by the log~2~-transformed expression level of each gene between *C. difficile*-infected and mock-infected conditions. Metatranscriptomic read abundances were normalized to their associated metagenomic coverage. Outliers were defined using linear correlation and a squared residual cutoff of 2. Distance of outliers to the x=y line were also calculated and represented in arbitrary units or AU. Colored indicate genus of origin, and gray areas denote genes with consistent expression between conditions (defined by outlier analysis). Antibiotic pretreatments; **(A)** Streptomycin-pretreated, **(B)** Cefoperazone-pretreated, and **(C)** Clindamycin-pretreated. 


**Supplementary Figure 1 | Impact of antibiotic pretreatment and *C. difficile* infection on cecal bacterial community structure** **(A)** Inverse-Simpson diversity of cecal communities from all pretreatment groups. Gray stars indicate significant difference from no antibiotic controls (all *p* < 0.001). Black stars denote within pretreatment group significant difference between mock and *C. difficile*-infected communities. Differences were calculated using Wilcoxon rank-sum test with Benjamini-Hochberg correction. NMDS ordinations of Bray-Curtis distances comparing the groups labeled in the bottom left of each plotting area. **(B)** Antibiotic-pretreated compared to Untreated controls and **(C)** comparison between only antibiotic pretreatment groups. **(D-F)** Within antibiotic pretreatment comparisons for the effect of *C. difficile* colonization on community structure. Significant differences and correlation coefficients for ordination analyses were found using ANOSIM.

**Supplementary Figure 2 | Relative concentrations of select metabolite groups in each group** Metabolites included in this analysis were chosen based on their previously published links to *C. difficile* physiology or susceptibility to infection. Groups are as follows; **(A)** Bile acids, **(B)** Carbohydrates, and **(C)** Amino Acids. Significant differences were determined by Wilcoxon rank-sum test with Benjamini-Hochberg correction.

**Supplementary Figure 3 | Effect of infection on the cecal metabolome across pretreatment groups** Pooled analysis of antibiotic pretreated animals only. **(A)** NMDS ordination of Bray-Curtis distances differentiating mock and *C. difficile*-infected metabolomes (*p* = 0.075). **(B)** Random forest classification results for metabolites that effectively distinguish infected and uninfected conditions. Shown are relative concentrations of the top 10 metabolites with the highest mean decrease in accuracy from internal cross-validation. Shown in the top right corner of each panel are the metabolite names and mean decrease accuracy of each. Significant differences in concentration between mock and *C. difficile*-infected groups were determined by Wilcoxon rank-sum test with Benjamini-Hochberg correction.

**Supplementary Figure 4 | Gene-level expression for each community compared to the level expressed in resistant mice for each gene** Murine cecal metatransctipromic sequencing results. Both reads and genes atributed to *C. difficile* were removed prior to analysis to focus on the changes in the community caused by infection. Shown are genes in each groups with the largest disparity from a metatranscriptome from a community resistant to *C. difficile* colonization. Panels are displayed as follows; Streptomycin pretreatment: **(A)** mock infection, **(B)** *C. difficile* infection. Cefoperazone pretreatment: **(C)** mock infection, **(D)** *C. difficile* infection. Clindamycin pretreatment: **(E)** mock infection, **(F)** *C. difficile* infection.

**Supplementary Table 1 | Chosen antibiotic pretreatment regimes** Antibiotic classes, mechanisms, and dosage information for each pretreatment.

**Supplementary Table 2 | High-throughput sequencing read counts and metagenomic assembly quality** Raw and curated read abundances for both metagenomic and metatranscriptomic sequencing efforts. Raw read curation steps are outlined in Materials & Methods. Metagenomic contig summary statistics reflect the quality of assembly for each group.

**Supplementary Table 3 | Normalized cDNA abundances for each antibiotic pretreated group** Final cDNA read abundances for each gene from the corresponding metagenomic assembly. Normalization reflects standardizing cDNA read abundances to both read length and target gene length as well as metagenomic coverage for each gene (also normalized).

**Supplementary Table 4 | Median scaled intensity of metabolites for each antibiotic pretreatment and infection group** 

**Supplementary Figure 5 | Heatmap comparing Resistant and Susceptible mice for all measured metabolites** Relative concentrations of metabolites in each animal across all expremerimental groups. Sample names are listed along the right side, and *C. difficile* susceptibility is along the left. Hierarchical clustering was perormed for each KEGG compound category included in the untargeted metabolomic analysis separately and are listed in order under the figure panel. 


### References
