

# Generate some rarefactoin curves for supplement
pdf(file='results/supplement/figures/figure_S3.pdf', width=11, height=6)
layout(matrix(c(1,2,3,
                4,5,6,
                7,8,9,
                10,11,12), nrow=4, ncol=3, byrow=TRUE))
par(mar=c(3,3,1,1), mgp=c(2,1,0))
test <- t(as.matrix(setNames(as.numeric(cef_raw_reads$cef_metaG_reads), as.character(rownames(cef_raw_reads)))))
rarecurve(test, sample=cef_size, label=FALSE)
legend('topleft', legend='Cefoperazone-pretreated Metagenome', pt.cex=0, bty='n', cex=0.9)
legend('bottomright', legend=cef_size, pt.cex=0, bty='n')
mtext('a', side=2, line=2, las=2, adj=1, padj=-8, cex=1.5, font=2)
test <- t(as.matrix(setNames(as.numeric(cef_raw_reads$cef_630_metaT_reads), as.character(rownames(cef_raw_reads)))))
rarecurve(test, sample=cef_size, label=FALSE)
legend('topleft', legend='Cefoperazone-pretreated Metatranscriptome (Infected)', pt.cex=0, bty='n', cex=0.9)
legend('bottomright', legend=cef_size, pt.cex=0, bty='n')
mtext('b', side=2, line=2, las=2, adj=1, padj=-8, cex=1.5, font=2)
test <- t(as.matrix(setNames(as.numeric(cef_raw_reads$cef_mock_metaT_reads), as.character(rownames(cef_raw_reads)))))
rarecurve(test, sample=cef_size, label=FALSE)
legend('topleft', legend='Cefoperazone-pretreated Metatranscriptome (Mock)', pt.cex=0, bty='n', cex=0.9)
legend('bottomright', legend=cef_size, pt.cex=0, bty='n')
mtext('c', side=2, line=2, las=2, adj=1, padj=-8, cex=1.5, font=2)
test <- t(as.matrix(setNames(as.numeric(clinda_raw_reads$clinda_metaG_reads), as.character(rownames(clinda_raw_reads)))))
rarecurve(test, sample=clinda_size, label=FALSE)
legend('topleft', legend='Clindamycin-pretreated Metagenome', pt.cex=0, bty='n', cex=0.9)
legend('bottomright', legend=clinda_size, pt.cex=0, bty='n')
mtext('d', side=2, line=2, las=2, adj=1, padj=-8, cex=1.5, font=2)
test <- t(as.matrix(setNames(as.numeric(clinda_raw_reads$clinda_630_metaT_reads), as.character(rownames(clinda_raw_reads)))))
rarecurve(test, sample=clinda_size, label=FALSE)
legend('topleft', legend='Clindamycin-pretreated Metatranscriptome (Infected)', pt.cex=0, bty='n', cex=0.9)
legend('bottomright', legend=clinda_size, pt.cex=0, bty='n')
mtext('e', side=2, line=2, las=2, adj=1, padj=-8, cex=1.5, font=2)
test <- t(as.matrix(setNames(as.numeric(clinda_raw_reads$clinda_mock_metaT_reads), as.character(rownames(clinda_raw_reads)))))
rarecurve(test, sample=clinda_size, label=FALSE)
legend('topleft', legend='Clindamycin-pretreated Metatranscriptome (Mock)', pt.cex=0, bty='n', cex=0.9)
legend('bottomright', legend=clinda_size, pt.cex=0, bty='n')
mtext('f', side=2, line=2, las=2, adj=1, padj=-8, cex=1.5, font=2)
test <- t(as.matrix(setNames(as.numeric(strep_raw_reads$strep_metaG_reads), as.character(rownames(strep_raw_reads)))))
rarecurve(test, sample=strep_size, label=FALSE)
legend('topleft', legend='Streptomycin-pretreated Metagenome', pt.cex=0, bty='n', cex=0.9)
legend('bottomright', legend=strep_size, pt.cex=0, bty='n')
mtext('g', side=2, line=2, las=2, adj=1, padj=-8, cex=1.5, font=2)
test <- t(as.matrix(setNames(as.numeric(strep_raw_reads$strep_630_metaT_reads), as.character(rownames(strep_raw_reads)))))
rarecurve(test, sample=strep_size, label=FALSE)
legend('topleft', legend='Streptomycin-pretreated Metatranscriptome (Infected)', pt.cex=0, bty='n', cex=0.9)
legend('bottomright', legend=strep_size, pt.cex=0, bty='n')
mtext('h', side=2, line=2, las=2, adj=1, padj=-8, cex=1.5, font=2)
test <- t(as.matrix(setNames(as.numeric(strep_raw_reads$strep_mock_metaT_reads), as.character(rownames(strep_raw_reads)))))
rarecurve(test, sample=strep_size, label=FALSE)
legend('topleft', legend='Streptomycin-pretreated Metatranscriptome (Mock)', pt.cex=0, bty='n', cex=0.9)
legend('bottomright', legend=strep_size, pt.cex=0, bty='n')
mtext('i', side=2, line=2, las=2, adj=1, padj=-8, cex=1.5, font=2)
test <- t(as.matrix(setNames(as.numeric(noabx_raw_reads$noabx_metaG_reads), as.character(rownames(noabx_raw_reads)))))
rarecurve(test, sample=noabx_size, label=FALSE)
legend('topleft', legend='No Antibiotics Metagenome', pt.cex=0, bty='n', cex=0.9)
legend('bottomright', legend=noabx_size, pt.cex=0, bty='n')
mtext('j', side=2, line=2, las=2, adj=1, padj=-8, cex=1.5, font=2)
test <- t(as.matrix(setNames(as.numeric(noabx_raw_reads$noabx_metaT_reads), as.character(rownames(noabx_raw_reads)))))
rarecurve(test, sample=noabx_size, label=FALSE)
legend('topleft', legend='No Antibiotics Metatranscriptome', pt.cex=0, bty='n', cex=0.9)
legend('bottomright', legend=noabx_size, pt.cex=0, bty='n')
mtext('k', side=2, line=2, las=2, adj=1, padj=-8, cex=1.5, font=2)
par(mar=c(0,0,0,0))
plot(0, type='n', ylim=c(-5,5), xlim=c(5,5), ylab='', xlab='', xaxt='n', yaxt='n', axes=FALSE)
rm(test, cef_size, clinda_size, strep_size, noabx_size)
dev.off()


