#!/bin/sh
#PBS -N map2cd630genome_RNA
#PBS -l nodes=1:ppn=8
#PBS -l walltime=500:00:00
#PBS -l mem=46gb
#PBS -j oe
#PBS -m abe
#PBS -V
#PBS -M mljenior@umich.edu
#PBS -q first
#PBS -A schloss_lab

echo "ncpus-2.pbs"
cat $PBS_NODEFILE
qstat -f $PBS_JOBID

cd $PBS_O_WORKDIR

NCPUS=`wc -l $PBS_NODEFILE | awk '{print $1}'`


# Align paired-end reads to Full length C. difficile 630 genome and plasmid
/home/mljenior/bin/bowtie2/bowtie2 -f --fr -p 8 -x /mnt/EXT/Schloss-data/matt/metatranscriptomes_HiSeq/dbs/cd630_genome/cd630_genome_plasmid_db \
	-1 /mnt/EXT/Schloss-data/matt/metatranscriptomes_HiSeq/final_reads/${transcriptome}.read1.pool.cut.trim.silva_filter.mus_filter.fasta \
	-2 /mnt/EXT/Schloss-data/matt/metatranscriptomes_HiSeq/final_reads/${transcriptome}.read2.pool.cut.trim.silva_filter.mus_filter.fasta \
	-S ${transcriptome}.pair.RNA_reads2cdf_genome.sam
samtools view -bS ${transcriptome}.pair.RNA_reads2cdf_genome.sam > ${transcriptome}.pair.RNA_reads2cdf_genome.bam
samtools sort ${transcriptome}.pair.RNA_reads2cdf_genome.bam ${transcriptome}.pair.RNA_reads2cdf_genome.sort
rm ${transcriptome}.pair.RNA_reads2cdf_genome.sam

# Align orphaned reads to Full length C. difficile 630 genome and plasmid
/home/mljenior/bin/bowtie2/bowtie2 -f -p 8 -x /mnt/EXT/Schloss-data/matt/metatranscriptomes_HiSeq/dbs/cd630_genome/cd630_genome_plasmid_db \
	-U /mnt/EXT/Schloss-data/matt/metatranscriptomes_HiSeq/final_reads/${transcriptome}.orphan.pool.cut.trim.silva_filter.mus_filter.fasta \
	-S ${transcriptome}.orphan.RNA_reads2cdf_genome.sam
samtools view -bS ${transcriptome}.orphan.RNA_reads2cdf_genome.sam > ${transcriptome}.orphan.RNA_reads2cdf_genome.bam
samtools sort ${transcriptome}.orphan.RNA_reads2cdf_genome.bam ${transcriptome}.orphan.RNA_reads2cdf_genome.sort
rm ${transcriptome}.orphan.RNA_reads2cdf_genome.sam

# Merge alignments and convert to idxstats format
samtools merge ${transcriptome}.RNA_reads2cdf_genome.sort.merge.bam ${transcriptome}.pair.RNA_reads2cdf_genome.sort.bam ${transcriptome}.orphan.RNA_reads2cdf_genome.sort.bam
samtools sort ${transcriptome}.RNA_reads2cdf_genome.sort.merge.bam ${transcriptome}.RNA_reads2cdf_genome.sort.merge.sort

# Remove PCR duplicates and sort
/mnt/EXT/Schloss-data/matt/bin/java1.8/bin/java -Xmx2g -jar /mnt/EXT/Schloss-data/matt/bin/picard-tools-1.119/MarkDuplicates.jar \
	INPUT=${transcriptome}.RNA_reads2cdf_genome.sort.merge.sort.bam \
	OUTPUT=${transcriptome}.RNA_reads2cdf_genome.sort.merge.sort.rmdup.bam \
	METRICS_FILE=${transcriptome}.RNA_reads2cdf_genome.sort.merge.sort.rmdup.metrics \
	AS=TRUE \
	VALIDATION_STRINGENCY=LENIENT \
	MAX_FILE_HANDLES_FOR_READ_ENDS_MAP=1000 \
	REMOVE_DUPLICATES=TRUE
samtools sort ${transcriptome}.RNA_reads2cdf_genome.sort.merge.sort.rmdup.bam ${transcriptome}.RNA_reads2cdf_genome.sort.merge.sort.rmdup.sort.bam
samtools index ${transcriptome}.RNA_reads2cdf_genome.sort.merge.sort.rmdup.sort.bam

# Calculate coverage metrics
bedtools genomecov -ibam ${transcriptome}.RNA_reads2cdf_genome.sort.merge.sort.rmdup.sort.bam.bam \
	-g /mnt/EXT/Schloss-data/matt/metatranscriptomes_HiSeq/dbs/cd630_genome/cdifficile630.genome.plasmid.fasta \
	-bga > ${transcriptome}.RNA_reads2cdf_genome.coverage.txt


echo "qsub working directory absolute is"
echo $PBS_O_WORKDIR
exit
