#!/bin/sh
#PBS -N preCONCOCT
#PBS -q fluxod
#PBS -l qos=flux
#PBS -l nodes=1:ppn=4,mem=32GB
#PBS -l walltime=300:00:00
#PBS -j oe
#PBS -V
#PBS -M mljenior@umich.edu
#PBS -A pschloss_fluxod

echo "ncpus-2.pbs"
cat $PBS_NODEFILE
qstat -f $PBS_JOBID
cd $PBS_O_WORKDIR

NCPUS=`wc -l $PBS_NODEFILE | awk '{print $1}'`


load module samtools

# Align paired-end reads to contigs
/nfs/turbo/schloss-lab/bin/bowtie2 -f --fr -x ${metagenome}. -1 /nfs/turbo/schloss-lab/mljenior/metagenomes_HiSeq/final_reads/${metagenome}/${metagenome}.read1.pool.cut.trim.mus_filtered.pick.fasta -2 /nfs/turbo/schloss-lab/mljenior/metagenomes_HiSeq/final_reads/${metagenome}/${metagenome}.read2.pool.cut.trim.mus_filtered.pick.fasta -S ${metagenome}.reads2contigs.concoct.sam
samtools faidx ${metagenome}.final.contigs.cut.fa
samtools view -bt ${metagenome}.final.contigs.cut.fa.fai ${metagenome}.reads2contigs.concoct.sam > ${metagenome}.reads2contigs.concoct.bam
rm ${metagenome}.reads2contigs.concoct.sam
samtools sort ${metagenome}.reads2contigs.concoct.bam ${metagenome}.reads2contigs.concoct.sort
samtools index ${metagenome}.reads2contigs.concoct.sort.bam

# Get coverage using BEDtools
/nfs/turbo/schloss-lab/mljenior/bin/bedtools2/bin/genomeCoverageBed -ibam ${metagenome}.reads2contigs.concoct.sort.bam -g ${metagenome}.final.contigs.cut.fa > ${metagenome}.final.contigs.coverage.txt

# Create coverage table
python /nfs/turbo/schloss-lab/bin/CONCOCT-0.3.2/scripts/gen_input_table.py --isbedfiles ${metagenome}.final.contigs.cut.fa  ${metagenome}.final.contigs.coverage.txt > ${metagenome}.coveragetable.tsv

# Create linkage table
python /nfs/turbo/schloss-lab/bin/CONCOCT-0.3.2/scripts/bam_to_linkage.py -m 8 --regionlength 500 --fullsearch ${metagenome}.final.contigs.cut.fa ${metagenome}.reads2contigs.concoct.sort.bam > ${metagenome}.linkagetable.tsv


echo "qsub working directory absolute is"
echo $PBS_O_WORKDIR
exit
