#!/bin/sh
#PBS -N read_curation
#PBS -q fluxod
#PBS -l qos=flux
#PBS -l nodes=1:ppn=4,mem=32GB
#PBS -l walltime=300:00:00
#PBS -j oe
#PBS -V
#PBS -M mljenior@umich.edu
#PBS -A pschloss_fluxod

echo "ncpus-2.pbs"
cat $PBS_NODEFILE
qstat -f $PBS_JOBID

cd $PBS_O_WORKDIR

NCPUS=`wc -l $PBS_NODEFILE | awk '{print $1}'`


python2.7 /nfs/turbo/pschloss/bin/cutadapt-1.9.1/bin/cutadapt -h

# Pool reads by sample
zcat *_R1_*.fastq.gz > ${reads}.read1.pool.fastq
zcat *_R2_*.fastq.gz > ${reads}.read2.pool.fastq

# Return some stats on the pooled fastqs
python /nfs/turbo/pschloss/bin/seq_stats.py ${reads}.read1.pool.fastq > ${reads}.trimming.summary
python /nfs/turbo/pschloss/bin/seq_stats.py ${reads}.read2.pool.fastq >> ${reads}.trimming.summary

# Cut residual Illumina adapters
python2.7 /nfs/turbo/pschloss/bin/cutadapt-1.9.1/bin/cutadapt --error-rate=0.1 --overlap=10 -a ${forward} -A ${reverse} -o ${reads}.read1.pool.cut.fastq -p ${reads}.read2.pool.cut.fastq ${reads}.read1.pool.fastq ${reads}.read2.pool.fastq
python /nfs/turbo/pschloss/bin/seq_stats.py ${reads}.read1.pool.cut.fastq >> ${reads}.trimming.summary
python /nfs/turbo/pschloss/bin/seq_stats.py ${reads}.read2.pool.cut.fastq >> ${reads}.trimming.summary
rm *.pool.fastq

# Quality trim the reads based on quality scores
/nfs/turbo/pschloss/matt/home/bin/sickle/sickle pe -f ${reads}.read1.pool.cut.fastq -r ${reads}.read2.pool.cut.fastq -t sanger -o ${reads}.read1.pool.cut.trim.fastq -p ${reads}.read2.pool.cut.trim.fastq -s ${reads}.orphan.pool.cut.trim.fastq -q 30 -l 21
rm *.cut.fastq

# Convert fastqs to fasta files
awk '{print ">" substr($0,2);getline;print;getline;getline}' ${reads}.read1.pool.cut.trim.fastq > ${reads}.read1.pool.cut.trim.fasta
awk '{print ">" substr($0,2);getline;print;getline;getline}' ${reads}.read2.pool.cut.trim.fastq > ${reads}.read2.pool.cut.trim.fasta
awk '{print ">" substr($0,2);getline;print;getline;getline}' ${reads}.orphan.pool.cut.trim.fastq > ${reads}.orphan.pool.cut.trim.fasta
rm *.trim.fastq 

# Record stats about the rest of the trimming
python /nfs/turbo/pschloss/bin/seq_stats.py ${reads}.read1.pool.cut.trim.fasta >> ${reads}.trimming.summary
python /nfs/turbo/pschloss/bin/seq_stats.py ${reads}.read2.pool.cut.trim.fasta >> ${reads}.trimming.summary
python /nfs/turbo/pschloss/bin/seq_stats.py ${reads}.orphan.pool.cut.trim.fasta >> ${reads}.trimming.summary

# Move files to final reads
mv *.pool.cut.trim.fasta /nfs/turbo/pschloss/matt/metagenomes_HiSeq/final_reads/${reads}/


echo "qsub working directory absolute is"
echo $PBS_O_WORKDIR
exit
